### PR TITLE
[MRG] Update what's new

### LINKED ISSUE #260: Release.rst
For release notes

### PR DESCRIPTION
**Pull request by jasmainak**
closes #260 

It's WIP. Since it's a little repetitive task, I'll finish it when I'm at the library later today.

We should make sure to update it before merging a pull request from now on.

### CODE REVIEW

**Comment by jasmainak** [comment]
@pavanramkumar I'm done here. Let me know if I missed something. Or feel free to push directly to this branch.

**Comment by jasmainak** [comment]
@pavanramkumar it's funny how we didn't add many features but I'm quite happy about the improved test coverage and the refactoring. I think it's a good move in the long run and makes the library more robust.

File: doc/whats_new.rst [hunk 1]
```diff
@@ -14,17 +14,25 @@ Current
...
+    - Changed stopping criteria for convergence (a threshold on the change in objective value) which
+      stopped too learly. The new criteria is a threshold on the norm of the gradient, by `Pavan Ramkumar`_. 
 
 API
 ~~~
 
-    - ...
+    - Refactored `GLM` class into two classes: `GLM` for fitting a single regularization parameter.
+      and `GLMCV` for fitting along a regularization path with cross-validation and selection of best
+      regularization parameter by `Pavan Ramkuamr`_ and `Mainak Jas`_.
```
Comment (pavanramkumar) [inline doc/whats_new.rst]: typo here in last name

File: doc/whats_new.rst [hunk 2]
```diff
@@ -42,12 +50,14 @@ Changelog
 	- Add multinomial link function by `Daniel Acuna`_
 	- Add benchmarks by `Vincinius Marques`_
 
-.. _Aid Idrizović: http:/github.com/the872
-.. _Daniel Acuna: http://klab.smpp.northwestern.edu/wiki/index.php5/Daniel_Acuna
-.. _Eva Dyer: <http:/github.com/evadyer
+.. _Aid Idrizović: https://github.com/the872
+.. _Daniel Acuna: https://klab.smpp.northwestern.edu/wiki/index.php5/Daniel_Acuna
```
Comment (pavanramkumar) [inline doc/whats_new.rst]: https://acuna.io/

File: doc/whats_new.rst [hunk 3]
```diff
@@ -42,12 +50,14 @@ Changelog
...
+.. _Eva Dyer: http:/github.com/evadyer
```
Comment (pavanramkumar) [inline doc/whats_new.rst]: http://evadyer.github.io/

File: doc/whats_new.rst [hunk 4]
```diff
@@ -42,12 +50,14 @@ Changelog
...
 .. _Hugo Fernandes: <http:/github.com/hugoguh
```
Comment (pavanramkumar) [inline doc/whats_new.rst]: https://github.com/hugoguh

File: doc/whats_new.rst [hunk 5]
```diff
@@ -42,12 +50,14 @@ Changelog
...
 .. _Mainak Jas: https://perso.telecom-paristech.fr/mjas/
-.. _Matt Antalek: <https://github.com/themantalope
-.. _Pavan Ramkumar: http://klab.smpp.northwestern.edu/wiki/index.php5/Pavan_Ramkumar
-.. _Titipat Achakulvisut: http:/github.com/titipata
-.. _Vinicius Marques: <http:/github.com/marquesVF
+.. _Matt Antalek: https://github.com/themantalope
+.. _Pavan Ramkumar: https://klab.smpp.northwestern.edu/wiki/index.php5/Pavan_Ramkumar
```
Comment (pavanramkumar) [inline doc/whats_new.rst]: https://github.com/pavanramkumar

File: doc/whats_new.rst [hunk 6]
```diff
@@ -14,17 +14,25 @@ Current
 Changelog
 ~~~~~~~~~
 
-    - Add `'gamma'` distribution with log link by `Mainak Jas`_.
+	- Add option for fitting `'probit'` distribution by `Ravi Garg`_.
```
Comment (pavanramkumar) [inline doc/whats_new.rst]: and `Pavan Ramkumar` (approximate gradients / hessians / loss for probit)

**Comment by pavanramkumar** [comment]
@jasmainak 

some more issues we solved, not sure where it belongs in the changelog:
- improved convergence testing
- gradient checks

**Comment by pavanramkumar** [comment]
thanks! LGTM once those two contribs are added.

**Comment by jasmainak** [comment]
Great, added.

**Pull request merged**